mmcv/model_zoo/hub.py

```diff
@@ -0,0 +1,117 @@
+                     checkpoint_path: str, json_path: str) -> None:
+    r"""Update the model zoo information in the json file.
+
+    The format of json file storing the model information should be as below:
+
+    .. code-block:: json
+
+    [
+        {
+            'name': str,
+            'config': str,
+            'details': str,
+            'path': str
+        }
+        ...
+    ]
+
+    Args:
+        name (str): The name of the model to be updated. Usually, it is
+            the name of the config.
+        config (str): The path of the model's config.
+            It should be a relative path of the repo that the model belongs to.
+        details (dict[str, float]): The detail information of the model,
+            including performance and latency.
+        checkpoint_path (str): The path of the checkpoint. It a path that can
+            be loaded by the load_checkpoint API implemented in MMCV.
+        json_path (str): The JSON file path to be updated.
+    """
+    ori_json = load(json_path)
+    # TODO: add check logics, including
+    # 1. whether the name already exists
+    # 2. the validity of path
+    ori_json.append(
+        dict(name=name, config=config, details=details, path=checkpoint_path))
+    dump(json_path, ori_json)
+
+
+def process_checkpoint(in_file: str, out_file: str) -> None:
+    """Process the checkpoint to make it ready for publish.
+
+    This function removes 'optimizer' in the checkpoint that is used for
+    resuming training. Then it adds hash to the output file.
+
+    Args:
+        in_file (str): Path of the original checkpoint to be converted.
+        out_file (str): Target path of the converted checkpoint.
+    """
+    checkpoint = torch.load(in_file, map_location='cpu')
+    # remove optimizer for smaller file size
+    if 'optimizer' in checkpoint:
+        del checkpoint['optimizer']
+    # if it is necessary to remove some sensitive data in checkpoint['meta'],
+    # add the code here.
+    torch.save(checkpoint, out_file)
+    sha = subprocess.check_output(['sha256sum', out_file]).decode()
+    if out_file.endswith('.pth'):
+        out_file_name = out_file[:-4]
+    else:
+        out_file_name = out_file
+    final_file = out_file_name + f'-{sha[:8]}.pth'
+    subprocess.Popen(['mv', out_file, final_file])
+
+
+def get_model_by_config(builder: callable,
+                        config: str,
+                        pretrained: bool = True) -> callable:
+    """Obtain the model given builder and config name.
+
+    Args:
+        builder (callable): The build function that builds model based
+            on the config.
+        config (str): The name of the model, or the path of config in the repo.
+        pretrained (bool, optional): Whether load the released pre-trained
+            model. Defaults to True.
+
+    Returns:
+        tuple: (model, details)
+    """
+
+    # TODO: support to load the json for only once
+    #
+
+
+def load_model(
+        repo_or_dir: str,
```

Comment:
Maybe task name + model name would be a better choice, for example when we want to build a model from a self_develop codebase,   we may not be able to infer which builder should be used.